pkg/apis/v1beta1/nodepool.go

```diff
@@ -0,0 +1,141 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"fmt"
+	"sort"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/ptr"
+)
+
+// NodePoolSpec is the top level provisioner specification. Provisioners
+// launch nodes in response to pods that are unschedulable. A single provisioner
+// is capable of managing a diverse set of nodes. Node properties are determined
+// from a combination of provisioner and pod scheduling constraints.
+type NodePoolSpec struct {
+	// Template contains the template of possibilities for the provisioning logic to launch a NodeClaim with.
+	// NodeClaims launched from this NodePool will often be further constrained than the template specifies.
+	// +optional
+	Template NodeClaimTemplate `json:"template,omitempty"`
+	// Deprovisioning contains the parameters that relate to Karpenter's deprovisioning logic
+	// +kubebuilder:default={"consolidationTTL": "15s","consolidationPolicy": "WhenUnderutilized", "expirationTTL": "90d"}
+	// +optional
+	Deprovisioning Deprovisioning `json:"deprovisioning"`
+	// Limits define a set of bounds for provisioning capacity.
+	// +optional
+	Limits Limits `json:"limits,omitempty"`
+	// Weight is the priority given to the provisioner during scheduling. A higher
+	// numerical weight indicates that this provisioner will be ordered
+	// ahead of other provisioners with lower weights. A provisioner with no weight
+	// will be treated as if it is a provisioner with a weight of 0.
+	// +kubebuilder:validation:Minimum:=1
+	// +kubebuilder:validation:Maximum:=100
+	// +optional
+	Weight *int32 `json:"weight,omitempty"`
+}
+
+type Deprovisioning struct {
+	// ConsolidationTTL is the duration the controller will wait
+	// before attempting to terminate nodes that are empty
+	// Termination due to under-utilization is disabled if this field is not set.
+	// This field is mutually exclusive to consolidation.enabled
```

Comment:
Do we need this line? There isn't any consolidation.enabled anymore.

Reply:
Good catch!